### PR TITLE
WTrackMenu: Create DlgTrackInfo/DlgTagFetcher only on demand

### PR DESCRIPTION
**Pull request by uklotzde**
Many unused instances of DlgTrackInfo are created by WTrackMenu. Noticed after adding a database query to the init code.

TODO(XXX): Initialize WTrackMenu lazily on-demand instead of eagerly.

### CODE REVIEW

File: src/widget/wtrackmenu.cpp [hunk 1]
```diff
@@ -1482,6 +1476,19 @@ void WTrackMenu::slotShowTrackInfo() {
     if (isEmpty()) {
         return;
     }
+    // Give a nullptr parent because otherwise it inherits our style which can
+    // make it unreadable. Bug #673411
+    m_pTrackInfo.reset(new DlgTrackInfo(
+            nullptr,
+            m_pConfig,
+            m_pTrackModel));
+    connect(m_pTrackInfo.get(),
```
Comment (daschuer) [inline src/widget/wtrackmenu.cpp:1485]: data() Get is new in QT 5.11

File: src/widget/wtrackmenu.cpp [hunk 2]
```diff
@@ -1482,6 +1476,19 @@ void WTrackMenu::slotShowTrackInfo() {
...
+            &QDialog::finished,
+            [this]() {
+                if (m_pTrackInfo.get() == sender()) {
```
Comment (daschuer) [inline src/widget/wtrackmenu.cpp:1488]: data() Get is new in QT 5.11

**Review by daschuer — CHANGES_REQUESTED** [review]
Great. This surprisingly fixes also the issue that the dialog does not pop up when opening it for a different track. 
Uu-Applied changes are lost in that case. This was also the case in the original version and I think this is a users fault. 

Just a build failure.

**Comment by uklotzde** [comment]
Also applies top DlgTagFetcher. Unfortunately many merge conflicts, need to force push. When I find some time...

**Review by daschuer — APPROVED** [review]
LGTM, Thank you.

**Pull request merged**